### PR TITLE
Clarify layer transparency keymap.md

### LINKED ISSUE #9048: Paragraph about transparency not clear
The second paragraph under [Layer Precedence and Transparency](https://docs.qmk.fm/#/keymap?id=layer-precedence-and-transparency) contains grammatical errors and is hard to understand.

### PR DESCRIPTION
**Pull request by nblyumberg**
## Description

Revised the layer transparency paragraph based on feedback from this [issue](https://github.com/qmk/qmk_firmware/issues/9048)

Resolves #9048

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/keymap.md [hunk 1]
```diff
@@ -73,8 +73,20 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
...
+         /  KC_TRNS  //          
+        /___________//   <--- Lower layer (KC_A)
+        /___________/  
+    
+    In the above scenario, the non-transparent keys on the higher layer would be usable, but whenever KC_TRNS (or equivalent) is defined the keycode (KC_A)) on the lower level would be used. 
+
+**Note:** Valid ways to denote transparency on a given layer:
+* `KC_TRNS`
+* `KC_TRANS` (alias)
+* `_______`  (alias)
```
Comment (fauxpark) [inline docs/keymap.md:87]:
```suggestion
* `KC_TRANSPARENT`
* `KC_TRNS` (alias)
* `_______` (alias)
```

File: docs/keymap.md [hunk 2]
```diff
@@ -73,8 +73,20 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
...
+
+These do not have their own keycodes and allow the processing to fall through to the lower layer in search of a keycode to process.
```
Comment (fauxpark) [inline docs/keymap.md:89]:
```suggestion
These keycodes allow the processing to fall through lower layers in search of a non-transparent keycode to process.
```

File: docs/keymap.md [hunk 3]
```diff
@@ -73,8 +73,20 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
...
 
-You can place `KC_TRANS` on overlay layer changes just part of layout to fall back on lower or base layer.
-Key with `KC_TRANS` (`KC_TRNS` and `_______` are the alias) doesn't has its own keycode and refers to lower valid layers for keycode, instead.
+           ____________        
+          /           /  <--- Higher layer
+         /  KC_TRNS  //          
+        /___________//   <--- Lower layer (KC_A)
+        /___________/  
+    
+    In the above scenario, the non-transparent keys on the higher layer would be usable, but whenever KC_TRNS (or equivalent) is defined the keycode (KC_A)) on the lower level would be used.
```
Comment (fauxpark) [inline docs/keymap.md:82]:
```suggestion
    In the above scenario, the non-transparent keys on the higher layer would be usable, but whenever `KC_TRNS` (or equivalent) is defined, the keycode (`KC_A`) on the lower level would be used.
```

File: docs/keymap.md [hunk 4]
```diff
@@ -73,8 +73,20 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
 ### Layer Precedence and Transparency
 Note that ***higher layer has higher priority on stack of layers***, namely firmware falls down from top layer to bottom to look up keycode. Once it spots keycode other than **`KC_TRNS`**(transparent) on a layer it stops searching and lower layers aren't referred.
 
-You can place `KC_TRANS` on overlay layer changes just part of layout to fall back on lower or base layer.
-Key with `KC_TRANS` (`KC_TRNS` and `_______` are the alias) doesn't has its own keycode and refers to lower valid layers for keycode, instead.
+           ____________        
+          /           /  <--- Higher layer
+         /  KC_TRNS  //          
+        /___________//   <--- Lower layer (KC_A)
+        /___________/
```
Comment (fauxpark) [inline docs/keymap.md:80]:
```suggestion
           ____________
          /           /  <--- Higher layer
         /  KC_TRNS  //
        /___________//   <--- Lower layer (KC_A)
        /___________/
```

File: docs/keymap.md (hunk 3 above)
Comment (spidey3) [reply docs/keymap.md:82]: You might want to add some text to clarify that the "lower" layers appear _higher_ up in the keymap.c file...

File: docs/keymap.md [hunk 5]
```diff
@@ -73,8 +73,20 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
 ### Layer Precedence and Transparency
 Note that ***higher layer has higher priority on stack of layers***, namely firmware falls down from top layer to bottom to look up keycode. Once it spots keycode other than **`KC_TRNS`**(transparent) on a layer it stops searching and lower layers aren't referred.
```
Comment (noroadsleft) [inline docs/keymap.md:74]: Some wordsmithing/grammar fixes.

```suggestion
Note that ***higher layers have higher priority within the stack of layers***, namely firmware falls down from top layer to bottom to look up keycode. Once it spots keycode other than **`KC_TRNS`** (transparent) on a layer it stops searching and lower layers aren't referred.
```

File: docs/keymap.md [hunk 6]
```diff
@@ -71,10 +71,22 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
 
 
 ### Layer Precedence and Transparency
-Note that ***higher layer has higher priority on stack of layers***, namely firmware falls down from top layer to bottom to look up keycode. Once it spots keycode other than **`KC_TRNS`**(transparent) on a layer it stops searching and lower layers aren't referred.
-
-You can place `KC_TRANS` on overlay layer changes just part of layout to fall back on lower or base layer.
-Key with `KC_TRANS` (`KC_TRNS` and `_______` are the alias) doesn't has its own keycode and refers to lower valid layers for keycode, instead.
+Note that ***higher layers have higher priority within the stack of layers***, namely firmware falls down from top layer to bottom to look up keycode. Once it spots keycode other than **`KC_TRNS`** (transparent) on a layer it stops searching and lower layers aren't referred.
```
Comment (noroadsleft) [inline docs/keymap.md:74]:
```suggestion
Note that ***higher layers have higher priority within the stack of layers***. The firmware works its way down from the highest active layers to look up keycodes. Once the firmware locates a keycode other than `KC_TRNS` (transparent) on an active layer, it stops searching, and lower layers aren't referenced.
```

File: docs/keymap.md [hunk 7]
```diff
@@ -71,10 +71,22 @@ On the other hand, you can change `layer_state` to overlay the base layer with o
...
+        /___________/
+    
+    In the above scenario, the non-transparent keys on the higher layer would be usable, but whenever `KC_TRNS` (or equivalent) is defined, the keycode (`KC_A`) on the lower level would be used.
+
+**Note:** Valid ways to denote transparency on a given layer:
+* `KC_TRANSPARENT`
+* `KC_TRNS` (alias)
+* `_______` (alias)
+
+These keycodes allow the processing to fall through lower layers in search of a non-transparent keycode to process.
```
Comment (noroadsleft) [inline docs/keymap.md:89]:
```suggestion
These keycodes allow the processing to fall through to lower layers in search of a non-transparent keycode to process.
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**